testutil/testdata/grpc_query.go

```diff
@@ -51,3 +60,28 @@ var _ types.UnpackInterfacesMessage = &TestAnyResponse{}
 func (m *TestAnyResponse) UnpackInterfaces(unpacker types.AnyUnpacker) error {
 	return m.HasAnimal.UnpackInterfaces(unpacker)
 }
+
+// QueryReq is a generic func to handle deterministic query requests.
+func QueryReq[request proto.Message, response proto.Message](
```

Comment:
```suggestion
func DeterministicIterations[request proto.Message, response proto.Message](
```

or `DeterministicQuery`. I would just put the word Deterministic inside it.